pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -27,16 +28,39 @@ import (
 // Node is a set of constraints, compatible pods, and possible instance types that could fulfill these constraints. This
 // will be turned into one or more actual node instances within the cluster after bin packing.
 type Node struct {
```

Comment:
Squinting at this, it might make sense to reuse constraints
```
type Node struct {
  Constraints
  InstanceTypeOptions
  Pods
  PodResources
  Daemons
  DaemonResources
}
```

Reply:
I didn't because of https://github.com/aws/karpenter/issues/1350